src/widget/wlibrarytableview.cpp

```diff
@@ -3,22 +3,30 @@
 #include <QFocusEvent>
 #include <QFontMetrics>
 #include <QHeaderView>
+#include <QMultiMap>
 #include <QPalette>
 #include <QScrollBar>
 
 #include "library/trackmodel.h"
 #include "moc_wlibrarytableview.cpp"
 #include "util/math.h"
 #include "widget/wskincolor.h"
+#include "widget/wtracktableviewheader.h"
 #include "widget/wwidget.h"
 
+namespace {
+// number of items to stop purging
+constexpr int kClearModelStatesLowWatermark = 1000;
+// number of items to start purging
+constexpr int kClearModelStatesHighWatermark = 1100;
```

Comment:
A wartermark is for me a sign, visible if you hold a piece of paper against light.   

How about 
* kModelStateCacheLowThreshold
* kModelStateCacheHighThreshold